cli/src/modules/message.rs

```diff
@@ -0,0 +1,157 @@
+                ),
+            ],
+            None,
+        )?;
+
+        Ok(())
+    }
+
+    async fn sign(self: Arc<Self>, ctx: Arc<KaspaCli>, kaspa_address: &str, message: &str) -> Result<()> {
+        let kaspa_address = Address::try_from(kaspa_address)?;
+        if kaspa_address.version != Version::PubKey {
+            return Err(Error::custom("Address not supported for message signing. Only supports PubKey addresses"));
+        }
+
+        let pm = PersonalMessage(message);
+        let privkey = self.get_address_private_key(&ctx, kaspa_address).await?;
+
+        let sig_result = sign_message(&pm, &privkey);
+
+        match sig_result {
+            Ok(signature) => {
+                let sig_hex = faster_hex::hex_string(signature.as_slice());
+                tprintln!(ctx, "Signature: {}", sig_hex);
+            }
+            Err(_) => {
+                return Err(Error::custom("Message signing failed"));
+            }
+        }
+
+        Ok(())
+    }
+
+    async fn verify(self: Arc<Self>, ctx: Arc<KaspaCli>, kaspa_address: &str, signature: &str, message: &str) -> Result<()> {
+        let kaspa_address = Address::try_from(kaspa_address)?;
+        if kaspa_address.version != Version::PubKey {
+            return Err(Error::custom("Address not supported for message signing. Only supports PubKey addresses"));
+        }
+
+        let pubkey = XOnlyPublicKey::from_slice(&kaspa_address.payload[0..32]).unwrap();
+
+        let mut signature_hex = [0u8; 64];
+        faster_hex::hex_decode(signature.as_bytes(), &mut signature_hex)?;
+
+        let pm = PersonalMessage(message);
+        let verify_result = verify_message(&pm, &signature_hex.to_vec(), &pubkey);
+
+        match verify_result {
+            Ok(()) => {
+                tprintln!(ctx, "Message verified successfully!");
+            }
+            Err(_) => {
+                return Err(Error::custom("Verification failed"));
+            }
+        }
+
+        Ok(())
+    }
+
+    async fn get_address_private_key(self: Arc<Self>, ctx: &Arc<KaspaCli>, kaspa_address: Address) -> Result<[u8; 32]> {
+        let account = ctx.wallet().account()?;
+
+        match account.account_kind() {
+            AccountKind::Bip32 => {
+                let (wallet_secret, payment_secret) = ctx.ask_wallet_secret(Some(&account)).await?;
+                let keydata = account.prv_key_data(wallet_secret).await?;
+                let mut keys = HashMap::new();
+                let account = account.clone().as_derivation_capable().expect("expecting derivation capable");
+
+                let (receive, change) = account.derivation().addresses_indexes(&[&kaspa_address])?;
+                let private_keys = account.create_private_keys(&keydata, &payment_secret, &receive, &change)?;
+                for (address, private_key) in private_keys {
+                    keys.insert(address.clone(), private_key.secret_bytes());
```

Comment:
Instead of insertion you can compare addresses and return it directly

Reply:
Fixed in [69020cb](https://github.com/kaspanet/rusty-kaspa/pull/280/commits/69020cbc0ef03a807a05554acb96ffca57ebb281)